login/service_login_blackbox_test.go

```diff
@@ -268,10 +283,21 @@ func (s *serviceLoginBlackBoxTest) runOauth2LoginEndToEnd() {
 
 	redirectedToURLRef, err = url.Parse(*redirectedTo)
 	require.NoError(s.T(), err)
-	require.Equal(s.T(), redirectURL, redirectedToURLRef.Scheme+"://"+redirectedToURLRef.Host+redirectedToURLRef.Path)
-	require.Equal(s.T(), s.Configuration.GetPublicOauthClientID(), redirectedToURLRef.Query()["api_client"][0])
-	require.Equal(s.T(), state, redirectedToURLRef.Query()["state"][0])
-	require.Equal(s.T(), returnedCode, redirectedToURLRef.Query()["code"][0])
+
```

Comment:
what about splitting this function in 2 parts: above to perform the "business" call, and below in 2 distinct functions to verify the result?

Reply:
There are way too many places where we are validating state and code, I tried considering having a function to validate that - but then it might end up being called again and again, to an extent that there it the code might actually not look any clean :)